enterprise/management/commands/add_exec_ed_exclusion_to_catalogs.py

```diff
@@ -0,0 +1,90 @@
+"""
+Django management command to add Exec Ed exclusion flag to catalogs
+"""
+
+import logging
+
+from django.core.management import BaseCommand
+
+from enterprise.models import EnterpriseCatalogQuery, EnterpriseCustomerCatalog
+from integrated_channels.utils import batch_by_pk
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Enumerate the catalog filters and add course_type exclusions where required
+    """
+
+    def handle(self, *args, **options):
+        for catalog_query_batch in batch_by_pk(EnterpriseCatalogQuery):
+            for catalog_query in catalog_query_batch:
+                logger.info(f'{catalog_query.id} {catalog_query.include_exec_ed_2u_courses}')
+
+                if catalog_query.include_exec_ed_2u_courses:
+                    logger.info(
+                        'add_exec_ed_exclusion_to_catalogs '
+                        f'query {catalog_query.id} already includes exec ed'
+                    )
+                    continue
+
+                if catalog_query.content_filter.get('course_type__exclude'):
+                    logger.info(
+                        'add_exec_ed_exclusion_to_catalogs '
+                        f'query {catalog_query.id} already references course_type__exclude somehow'
+                    )
+                    continue
+
+                if catalog_query.content_filter.get('course_type'):
+                    logger.info(
+                        'add_exec_ed_exclusion_to_catalogs '
+                        f'query {catalog_query.id} already references course_type somehow'
+                    )
+                    continue
+
+                if catalog_query.content_filter.get('aggregation_key'):
```

Comment:
why is this an issue?

Reply:
meaning, for the sake of tagging query filters for exec ed, why do we care about aggregation key? Because it's selective on a lower lever than course type?

Reply:
the rationale is that if they're selecting specific courses they already know which are which